crates/core/tedge_agent/src/software_manager/actor.rs

```diff
@@ -77,10 +79,15 @@ impl Actor for SoftwareManagerActor {
         let operation_logs = OperationLogs::try_new(self.config.log_dir.clone().into())
             .map_err(SoftwareManagerError::FromOperationsLogs)?;
 
+        let sudo: Option<PathBuf> = match which(SUDO) {
+            Ok(_) => Some(SUDO.into()),
+            Err(_) => None,
+        };
+
```

Comment:
Use the path returned by `which`.

```suggestion
        let sudo = which(SUDO).ok();
```